src/cross_sections.cpp

```diff
@@ -0,0 +1,433 @@
+// Library methods
+//==============================================================================
+
+Library::Library(pugi::xml_node node, const std::string& directory)
+{
+  // Get type of library
+  if (check_for_node(node, "type")) {
+    auto type = get_node_value(node, "type");
+    if (type == "neutron") {
+      type_ = Type::neutron;
+    } else if (type == "thermal") {
+      type_ = Type::thermal;
+    } else if (type == "photon") {
+      type_ = Type::photon;
+    } else if (type == "wmp") {
+      type_ = Type::wmp;
+    } else {
+      fatal_error("Unrecognized library type: " + type);
+    }
+  } else {
+    fatal_error("Missing library type");
+  }
+
+  // Get list of materials
+  if (check_for_node(node, "materials")) {
+    materials_ = get_node_array<std::string>(node, "materials");
+  }
+
+  // determine path of cross section table
+  if (!check_for_node(node, "path")) {
+    fatal_error("Missing library path");
+  }
+  std::string path = get_node_value(node, "path");
+
+  if (starts_with(path, "/")) {
+    path_ = path;
+  } else if (ends_with(directory, "/")) {
+    path_ = directory + path;
+  } else {
+    path_ = directory + "/" + path;
+  }
+
+  if (!file_exists(path_)) {
+    warning("Cross section library " + path_ + " does not exist.");
+  }
+}
+
+//==============================================================================
+// Non-member functions
+//==============================================================================
+
+void read_cross_sections_xml()
+{
+  pugi::xml_document doc;
+  std::string filename = settings::path_input + "materials.xml";
+#ifdef DAGMC
+  std::string s;
+  bool found_uwuw_mats = false;
+  if (settings::dagmc) {
+    found_uwuw_mats = get_uwuw_materials_xml(s);
+  }
+
+  if (found_uwuw_mats) {
+    // if we found uwuw materials, load those
+    doc.load_file(s.c_str());
+  } else {
+#endif
+  // Check if materials.xml exists
+  if (!file_exists(filename)) {
+    fatal_error("Material XML file '" + filename + "' does not exist.");
+  }
+  // Parse materials.xml file
+  doc.load_file(filename.c_str());
+#ifdef DAGMC
+  }
+#endif
+
+  auto root = doc.document_element();
+
+  // Find cross_sections.xml file -- the first place to look is the
+  // materials.xml file. If no file is found there, then we check the
+  // OPENMC_CROSS_SECTIONS environment variable
+  if (!check_for_node(root, "cross_sections")) {
+    // No cross_sections.xml file specified in settings.xml, check
+    // environment variable
+    if (settings::run_CE) {
+      char* envvar = std::getenv("OPENMC_CROSS_SECTIONS");
+      if (!envvar) {
+        fatal_error("No cross_sections.xml file was specified in "
+          "materials.xml or in the OPENMC_CROSS_SECTIONS"
+          " environment variable. OpenMC needs such a file to identify "
+          "where to find data libraries. Please consult the"
+          " user's guide at https://openmc.readthedocs.io for "
```

Comment:
This URL is valid, though the docs are also available at https://docs.openmc.org/, so I'm wondering if the URL should be changed.